src/zcash/address/unified.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2021 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef ZCASH_ZCASH_ADDRESS_UNIFIED_H
+#define ZCASH_ZCASH_ADDRESS_UNIFIED_H
+
+#include "zip32.h"
+#include "bip44.h"
+
+namespace libzcash {
+
+class ZcashdUnifiedSpendingKey;
+class ZcashdUnifiedFullViewingKey;
+
+class ZcashdUnifiedAddress {
+private:
+    diversifier_index_t diversifier_index;
+    std::optional<CKeyID> transparentAddress;
+    std::optional<SaplingPaymentAddress> saplingAddress;
+
+    friend class ZcashdUnifiedFullViewingKey;
+
+    ZcashdUnifiedAddress() {}
+public:
+    const std::optional<CKeyID>& GetTransparentAddress() const {
+        return transparentAddress;
+    }
+
+    const std::optional<SaplingPaymentAddress>& GetSaplingPaymentAddress() const {
+        return saplingAddress;
+    }
+};
+
+class ZcashdUnifiedFullViewingKey {
+private:
+    std::optional<CExtPubKey> transparentKey;
+    std::optional<SaplingExtendedFullViewingKey> saplingKey;
+
+    ZcashdUnifiedFullViewingKey() {}
+
+    friend class ZcashdUnifiedSpendingKey;
+public:
+    const std::optional<CExtPubKey>& GetTransparentKey() const {
+        return transparentKey;
+    }
+
+    const std::optional<SaplingExtendedFullViewingKey>& GetSaplingExtendedFullViewingKey() const {
+        return saplingKey;
+    }
+
+    std::optional<ZcashdUnifiedAddress> Address(diversifier_index_t j) const;
+
+    std::pair<ZcashdUnifiedAddress, diversifier_index_t> FindAddress(diversifier_index_t j) const {
+        auto addr = Address(j);
+        while (!addr.has_value()) {
+            if (!j.increment())
+                throw std::runtime_error(std::string(__func__) + ": diversifier index overflow.");;
+            addr = Address(j);
+        }
+        return std::make_pair(addr.value(), j);
+    }
+};
+
+class ZcashdUnifiedSpendingKey {
+private:
+    libzcash::AccountId accountId;
+    std::optional<CExtKey> transparentKey;
+    std::optional<SaplingExtendedSpendingKey> saplingKey;
+
+    ZcashdUnifiedSpendingKey() {}
+public:
+    static std::optional<std::pair<ZcashdUnifiedSpendingKey, HDKeyPath>> ForAccount(
+            const HDSeed& seed,
+            uint32_t bip44CoinType,
+            libzcash::AccountId accountId);
+
+    const std::optional<CExtKey>& GetTransparentKey() const {
+        return transparentKey;
+    }
+
+    const std::optional<SaplingExtendedSpendingKey>& GetSaplingExtendedSpendingKey() const {
+        return saplingKey;
+    }
+
+    ZcashdUnifiedFullViewingKey ToFullViewingKey() const;
+};
+
+std::optional<unsigned long> ParseHDKeypathAccount(uint32_t purpose, uint32_t coinType, const std::string& keyPath);
+
```

Comment:
AFAICT this is mis-copied, as the original declaration is still in `zip32.h` and its definition is still in `zip32.cpp`:
```suggestion
```